src/Workspaces/Remote/ServiceHub/Host/RemoteWorkspace.cs

```diff
@@ -106,6 +120,46 @@ public AssetProvider CreateAssetProvider(Checksum solutionChecksum, SolutionAsse
             bool updatePrimaryBranch,
             Func<Solution, ValueTask<T>> implementation,
             CancellationToken cancellationToken)
```

Comment:
@dibarbet trying this again with a different approach.  instead of adding a new 'pinned solution' type and having the callers own pinning things, i matain the original pattern as before where the pinning happens here and the 'impl' function is called while the solution is pinned by us.

The difference now is that i allow the callback to be a streaming one.  so we pin hte solution here like normal, then stream the results, then unpin when tehy're done.

THe normal case of a callback that just produces one value is then mapped to the streaming form by passing in a stream that only returns one value.

Reply:
this in general looks fine - but it also looked fine last time.  Is there any way we can unit tests this to ensure that the right things are happening to the pinned solution on both the client and servicehub side?  I'd be much more confident in changes here if we could.